source/Digitisers/src/DDPlanarDigiProcessor.cc

```diff
@@ -308,10 +348,41 @@ void DDPlanarDigiProcessor::processEvent( LCEvent * evt ) {
           continue; 
         }
       }
+
+      //***************************************************************
+      // Smear time of the hit and apply the time window cut if needed
+      //***************************************************************
       
-      //**************************************************************************
-      // Try to smear the hit but ensure the hit is inside the sensitive region
-      //**************************************************************************
+      // Smearing time of the hit
+      float resT = _resT.size() > 1 ? _resT.at(layer) : _resT.at(0); 
+      double tSmear  = resT == 0.0 ? 0.0 : gsl_ran_gaussian( _rng, resT );
```

Comment:
```suggestion
      double tSmear  = resT > 0.0 : gsl_ran_gaussian( _rng, resT ) ? 0.0;
```
I guess it is possible that one would like to disable smearing for some layers, so we need to keep testing